src/Analyzers/CSharp/Analyzers/UseCollectionExpression/CSharpUseCollectionExpressionForArrayDiagnosticAnalyzer.cs

```diff
@@ -15,22 +15,18 @@
 namespace Microsoft.CodeAnalysis.CSharp.UseCollectionExpression;
 
 [DiagnosticAnalyzer(LanguageNames.CSharp)]
-internal sealed partial class CSharpUseCollectionExpressionForArrayDiagnosticAnalyzer
-    : AbstractCSharpUseCollectionExpressionDiagnosticAnalyzer
+internal sealed partial class CSharpUseCollectionExpressionForArrayDiagnosticAnalyzer()
+    : AbstractCSharpUseCollectionExpressionDiagnosticAnalyzer(
+        IDEDiagnosticIds.UseCollectionExpressionForArrayDiagnosticId,
+        EnforceOnBuildValues.UseCollectionExpressionForArray)
 {
-    public CSharpUseCollectionExpressionForArrayDiagnosticAnalyzer()
-        : base(IDEDiagnosticIds.UseCollectionExpressionForArrayDiagnosticId,
-               EnforceOnBuildValues.UseCollectionExpressionForArray)
+    protected override void InitializeWorker(CodeBlockStartAnalysisContext<SyntaxKind> context, INamedTypeSymbol? expressionType)
     {
+        context.RegisterSyntaxNodeAction(context => AnalyzeArrayInitializerExpression(context, expressionType), SyntaxKind.ArrayInitializerExpression);
```

Comment:
>RegisterSyntaxNodeAction [](http://example.com/codeflow?start=16&length=24)

Super dumb question here. It doesn't look like RegisterSyntaxNodeAction allows for the trick where you can pass an additional param to allow for a static closure to be used. Would it make sense for it to?